### ISSUE TITLE
IndieAuth endpoint does not support multi-user

### ISSUE THREAD

**Issue by paulcmal** [idno/Known#1499]
## While trying to do this:

Logging into [indieauth.com](https://indieauth.com) using my profile URL on a multi-user Known.

## I encountered this error:

« Endpoint did not acknowledge it is an authorization endpoint. The endpoint should return an "IndieAuth: authorization_endpoint" header. »

![Indieauth.com error](https://pix.unixcorn.org/image/1472055902/large.jpg)

## Some other notes:

Someone suggested on IRC to try enabling single-user mode (with multiple users) as a fix. It did not change anything.

This is related to issue #1351 and PR #1353.

**Comment by kylewm** [comment]
I don't think this is related to mutli-user. The issue seems to be that /indieauth/auth redirects to /session/login (when you're not logged in) which doesn't report the header indieauth.com expects.

**Comment by paulcmal** [comment]
> The issue seems to be that /indieauth/auth redirects to /session/login (when you're not logged in)

[You're right](https://github.com/idno/Known/blob/38eb3bcf5f743110b8bfad8a76ae74b5cfe0fcbd/IdnoPlugins/IndiePub/Pages/IndieAuth/Auth.php#L14).

> doesn't report the header indieauth.com expects

I finally found [more info](https://indieweb.org/authorization-endpoint#Identification_header) about this expected header: `IndieAuth: authorization_endpoint`. From what I could see, it's not documented neither on indieauth.com, nor on the IndieAuth/IndieAuthProtocol Indieweb wiki pages.

I couldn't find trace of this header in the codebase (in your commit #1114 or now). I'm assuming it was never implemented.

I did try to login manually by going to /indieauth/auth and providing by hand client_id, me, and redirect_uri and it did ask me to approve the request and redirected me to indieweb.org as expected, refusing my authentication as expected (as there was no valid token involved).

Taking a look at how @aaronpk's [auth](https://aaronparecki.com/auth) works, it seems to respond with 200 whether no parameter is provided, or further authentication is needed, and always with a valid IndieAuth header.

I'm going to do some testing tomorrow to see if I can get anywhere by just adding the header (through nginx) and, if needed, changing error code 403 to 200.

Thank you for your insight!
